util/types/mytime.go

```diff
@@ -182,6 +182,11 @@ func calcDaynr(year, month, day int) int {
 	return delsum + year/4 - temp
 }
 
+// DateDiff calculates number of days between two days.
+func DateDiff(startTime, endTime TimeInternal) (int, error) {
```

Comment:
The `err` doesn't catch on line 108 of the `expression/builtin_time.go` file. I think `DateDiff` needn't return an error.